gobblin-metrics/src/main/java/gobblin/metrics/KafkaReporter.java

```diff
@@ -0,0 +1,426 @@
+   */
+  public static String makeName(String name, String... path) {
+    final StringBuilder sb = new StringBuilder(name);
+    for (String part: path) {
+      sb.append('.').append(part);
+    }
+    return sb.toString();
+  }
+
+  protected KafkaReporter(Builder<?> builder) {
+    super(builder.registry, builder.name, builder.filter,
+        builder.rateUnit, builder.durationUnit);
+
+    lastSerializeExceptionTime = 0;
+
+    this.tags = builder.tags;
+
+    Properties props = new Properties();
+
+    props.put("metadata.broker.list", builder.brokers);
+    props.put("serializer.class", "kafka.serializer.DefaultEncoder");
+    props.put("request.required.acks", "1");
+
+    config = new ProducerConfig(props);
+    producer = new Producer<String, byte[]>(config);
+    this.topic = builder.topic;
+
+  }
+
+  @Override
+  public void close() {
+    super.close();
+    producer.close();
+  }
+
+  /**
+   * Serializes metrics and pushes the byte arrays to Kafka.
+   * Uses the serialize* methods in {@link gobblin.metrics.KafkaReporter}
+   * @param gauges
+   * @param counters
+   * @param histograms
+   * @param meters
+   * @param timers
+   */
+  @Override
+  public void report(SortedMap<String, Gauge> gauges, SortedMap<String, Counter> counters,
+      SortedMap<String, Histogram> histograms, SortedMap<String, Meter> meters, SortedMap<String, Timer> timers) {
+
+    List<KeyedMessage<String, byte[]>> messages = new ArrayList<KeyedMessage<String, byte[]>>();
+
+    for( Entry<String, Gauge> gauge : gauges.entrySet()) {
+      messages.addAll(toKeyedMessages(serializeGauge(gauge.getKey(), gauge.getValue())));
+    }
+
+    for ( Entry<String, Counter> counter : counters.entrySet()) {
+      messages.addAll(toKeyedMessages(serializeCounter(counter.getKey(), counter.getValue())));
+    }
+
+    for( Entry<String, Histogram> histogram : histograms.entrySet()) {
+      messages.addAll(toKeyedMessages(serializeSnapshot(histogram.getKey(), histogram.getValue().getSnapshot())));
+      messages.addAll(
+          toKeyedMessages(serializeSingleValue(histogram.getKey(), histogram.getValue().getCount(), "count")));
+    }
+
+    for ( Entry<String, Meter> meter : meters.entrySet()) {
+      messages.addAll(toKeyedMessages(serializeMetered(meter.getKey(), meter.getValue())));
+    }
+
+    for ( Entry<String, Timer> timer : timers.entrySet()) {
+      messages.addAll(toKeyedMessages(serializeSnapshot(timer.getKey(), timer.getValue().getSnapshot())));
+      messages.addAll(toKeyedMessages(serializeMetered(timer.getKey(), timer.getValue())));
+      messages.addAll(toKeyedMessages(serializeSingleValue(timer.getKey(), timer.getValue().getCount(), "count")));
+    }
+
+    producer.send(messages);
+
+  }
+
+  /**
+   * Serialize a {@link com.codahale.metrics.Gauge}
+   * @param name
+   * @param gauge
+   * @return List of byte arrays for each metric derived from the gauge
+   */
+  protected List<byte[]> serializeGauge(String name, Gauge gauge) {
```

Comment:
For consistency with the other serialize methods mostly. And in case an override of this method wants to send more than one line for a gauge.